### PR TITLE
Harden AsyncDnsResolverIntegrationSpec

### PR DESCRIPTION
**Pull request by chbatey**
* Ensure that the port is available on TCP and UDP
* Don't listen on ipv6 (fails on jenkins sometimes)

### CODE REVIEW

File: akka-actor-tests/src/test/bind/etc/named.conf.options [hunk 1]
```diff
@@ -21,6 +21,6 @@ options {
 	dnssec-validation auto;
 
 	auth-nxdomain no;    # conform to RFC1035
-	listen-on-v6 { any; };
+	listen-on-v6 { none; };
```
Comment (raboof) [inline akka-actor-tests/src/test/bind/etc/named.conf.options]: I can't believe IPv6 is still problematic ;) :+1:

File: akka-actor-tests/src/test/scala/akka/io/dns/AsyncDnsResolverIntegrationSpec.scala [hunk 2]
```diff
@@ -152,6 +154,24 @@ class AsyncDnsResolverIntegrationSpec extends AkkaSpec(
 
   }
 }
+
 object AsyncDnsResolverIntegrationSpec {
-  lazy val dockerDnsServerPort = SocketUtil.temporaryLocalPort()
+  lazy val dockerDnsServerPort: Int = {
+    def freePort(tries: Int): Int = {
```
Comment (raboof) [inline akka-actor-tests/src/test/scala/akka/io/dns/AsyncDnsResolverIntegrationSpec.scala]: Sounds like something that would be useful in `SocketUtil`, perhaps move it?

**Review by raboof — APPROVED** [review]

File: akka-actor-tests/src/test/scala/akka/io/dns/AsyncDnsResolverIntegrationSpec.scala (hunk 2 above)
Comment (patriknw) [reply akka-actor-tests/src/test/scala/akka/io/dns/AsyncDnsResolverIntegrationSpec.scala]: I agree, note that it has some stuff for RANDOM_LOOPBACK_ADDRESS

File: akka-actor-tests/src/test/scala/akka/io/dns/AsyncDnsResolverIntegrationSpec.scala [hunk 3]
```diff
@@ -152,6 +154,24 @@ class AsyncDnsResolverIntegrationSpec extends AkkaSpec(
...
+      if (tries == 0) {
+        throw new RuntimeException("Unable to find a port that is free for tcp and udp")
+      }
+      val tcpPort = SocketUtil.temporaryLocalPort()
+      val ds: DatagramSocket = DatagramChannel.open().socket()
+      try {
+        ds.bind(new InetSocketAddress("localhost", tcpPort))
```
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/io/dns/AsyncDnsResolverIntegrationSpec.scala]: SocketUtil is using 127.0.0.1 or RANDOM_LOOPBACK_ADDRESS stuff

**Comment by chbatey** [comment]
Moved to SocketUtil :+1:

**Comment by chbatey** [comment]
Bah formatting

**Review by jrudolph — APPROVED** [review]
LGTM (at least not worse than before, the "temporary free port" mechanism is still quite unreliable if anything else runs on the same machine at the same time...)

**Pull request merged**